app/api/search/search.js

```diff
@@ -147,96 +161,193 @@ const processResponse = (response) => {
+    documentsQuery.aggregations(aggregations, dictionaries);
 
-      documentsQuery.filterMetadataByFullText(textSearchFilters);
-      documentsQuery.filterMetadata(filters);
-      documentsQuery.aggregations(aggregations, dictionaries);
+    if (query.geolocation) {
+      searchGeolocation(documentsQuery, filteringTypes, templates);
+    }
+    return elastic.search({ index: elasticIndex, body: documentsQuery.query() })
+    .then(processResponse)
+    .catch(() => {
+      throw createError('Query error', 400);
+    });
+  });
+};
 
-      if (query.geolocation) {
-        searchGeolocation(documentsQuery, filteringTypes, templates);
+const determineInheritedProperties = templates => templates.reduce((memo, template) => {
+  const inheritedProperties = memo;
+  template.properties.forEach((property) => {
+    if (property.type === 'relationship' && property.inherit) {
+      const contentTemplate = templates.find(t => t._id.toString() === property.content.toString());
+      const inheritedProperty = contentTemplate.properties.find(
+        p => p.type === 'geolocation' && p._id.toString() === property.inheritProperty.toString()
+      );
+      if (inheritedProperty) {
+        inheritedProperties[template._id.toString()] = inheritedProperties[template._id.toString()] || {};
+        inheritedProperties[template._id.toString()][property.name] = { base: property, target: inheritedProperty };
       }
-      return elastic.search({ index: elasticIndex, body: documentsQuery.query() })
-      .then(processResponse)
-      .catch(() => {
-        throw createError('Query error', 400);
-      });
+    }
+  });
+  return inheritedProperties;
+}, {});
+
+const whatToFetchByTemplate = (baseResults, templatesInheritedProperties) => {
+  const toFetchByTemplate = {};
+
+  baseResults.rows.forEach((row) => {
+    Object.keys(row.metadata || {}).forEach((name) => {
+      if (Object.keys(templatesInheritedProperties[row.template.toString()] || []).includes(name)) {
+        const inheritedProperty = templatesInheritedProperties[row.template.toString()][name];
+        const template = inheritedProperty.base.content;
+        toFetchByTemplate[template] = toFetchByTemplate[template] || { entities: [], properties: [] };
+        toFetchByTemplate[template].entities = toFetchByTemplate[template].entities.concat(row.metadata[name]);
+        if (!toFetchByTemplate[template].properties.includes(inheritedProperty.target.name)) {
+          toFetchByTemplate[template].properties.push(inheritedProperty.target.name);
+        }
+      }
+    });
+  });
+
+  return toFetchByTemplate;
+};
+
+const getInheritedEntitiesData = async (toFetchByTemplate, language) => Promise.all(
+  Object.keys(toFetchByTemplate).map(t => entities.get(
+    { language, sharedId: { $in: toFetchByTemplate[t].entities } },
+    { ...toFetchByTemplate[t].properties.reduce((memo, n) => Object.assign(memo, { [`metadata.${n}`]: 1 }), {}), sharedId: 1 }
```

Comment:
is sharedId hardcoded to 1 intended?

Reply:
This is the 'project' part of the get.  It's to just fetch the shared ID and the expected metadata and not all the other info in entities.  Its:
entities.get({ _id: { $in: [1,2,3]} }, {'metadata.prop1': 1, 'metadata.prop2': 1, sharedId: 1})